src/main/scala/com/imaginea/activegrid/core/models/SharedSessionCache.scala

```diff
@@ -0,0 +1,63 @@
+package com.imaginea.activegrid.core.models
+
+import akka.actor.Actor
+import akka.actor.ActorRef
+import akka.actor.Props
+import akka.cluster.Cluster
+import akka.cluster.ddata.{DistributedData, LWWMap, LWWMapKey, Replicator}
+import akka.cluster.ddata.Replicator._ //scalastyle:ignore underscore.import
+import com.typesafe.scalalogging.Logger
+import scala.concurrent.duration._  //scalastyle:ignore underscore.import
+import SharedSessionCache._ //scalastyle:ignore underscore.import
+import org.slf4j.LoggerFactory
+/**
+  * Created by shareefn on 3/3/17.
+  */
+object SharedSessionCache {
+
+  val props: Props = Props[SharedSessionCache]
+
+  case class Request(key: String, replyTo: ActorRef)
+  case class PutSession(key: String, value: TerminalSession)
+  case class GetSession(key: String)
+  case class RemoveSession(key: String)
+
+  val timeout = 10.seconds
+  val readMajority = ReadMajority(timeout)
+  val writeMajority = WriteMajority(timeout)
+}
+
+class SharedSessionCache extends Actor {
+
+  val replicator = DistributedData(context.system).replicator
+  val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+  implicit val cluster = Cluster(context.system)
+
+  def dataKey(entryKey: String): LWWMapKey[TerminalSession] =
+    LWWMapKey(entryKey)
+
+  def receive: PartialFunction[Any, Unit] = {
+    case PutSession(key, value) =>
+      logger.info(s"inserting terminal session in shared map with key: $key")
+      replicator ! Replicator.Update(dataKey(key), LWWMap(), writeMajority)(_ + (key -> value))
+    case RemoveSession(key) =>
+      logger.info(s"removing terminal session in shared map with key: $key")
+      replicator ! Replicator.Update(dataKey(key), LWWMap(), writeMajority)(_ - key)
+    case GetSession(key) =>
```

Comment:
We discussed that Siva should use ask in his code. But I don't remember anything to be changed here.
Could you tell me what needs to be changed and I will change it.